notificationdemo/source/android/app/src/main/java/org/kaaproject/kaa/demo/notification/storage/TopicStorage.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright 2014-2016 CyberVision, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kaaproject.kaa.demo.notification.storage;
+
+import android.content.Context;
+import android.preference.PreferenceManager;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+
+import org.kaaproject.kaa.demo.notification.entity.TopicPojo;
+import org.kaaproject.kaa.demo.notification.util.NotificationConstants;
+import org.kaaproject.kaa.demo.notification.util.TopicHelper;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Save and load topic list from {@link android.content.SharedPreferences}.
+ * Use {@link Gson} for format topic list as json string.
+ * Change topics for subscribing/unsubscribing events.
+ * For more information use {@see <a href="https://github.com/google/gson/blob/master/GsonDesignDocument.md">Gson docs</a>}
+ * <p>
+ * Tip: don't use Shared Preferences for storing such information. This class used only for example
+ */
+public class TopicStorage {
```

Comment:
Why do you need this class and associated logic? Can't you just use this method: http://kaaproject.github.io/kaa/doc/client-java-core/latest/org/kaaproject/kaa/client/GenericKaaClient.html#getTopics--